### ISSUE TITLE
Quad ordering is incorrect after material downgrading, causing problems with certain block models

### ISSUE THREAD

**Issue by joe-vettek** [jellysquid3/Sodium#2868]
### Bug Description

In vanilla MC, Mojang allowed modders to overlay textures by means of isotropic blocks, without being affected by z-depth.
An example is the grass block model, which makes layered tinting possible and more flexible.
However, in Sodium, although side coverage is still supported, There is a weird sorting problem here.
![image](https://github.com/user-attachments/assets/f5319c28-2db7-434c-a5f2-f0cd54094abf)
![image](https://github.com/user-attachments/assets/e8c9ba9e-7ebb-420d-8882-6c31d5819eca)

### Reproduction Steps

1. open minecraft and use version 1.21.1
2. use Sodium 
3. use the test resourcepack
[TESTPack.zip](https://github.com/user-attachments/files/17641503/TESTPack.zip)
4. place oak leaves or blue stained glass

### Log File

None log, because it not crash.
[debug.log](https://github.com/user-attachments/files/17641594/debug.log)

### Crash Report

[crash-2024-11-06_12.14.25-client.txt](https://github.com/user-attachments/files/17641604/crash-2024-11-06_12.14.25-client.txt)

**Comment by jellysquid3** [comment]
Does this issue happen _without_ Iris?

**Comment by douira** [comment]
I can reproduce without Iris on `dev`

**Comment by douira** [comment]
For reference, the result vanilla produces is oak leaves that look like a snow block.

![Screenshot 2024-11-06 at 05 27 31](https://github.com/user-attachments/assets/b70fd9bf-6b16-4a14-8462-ac8b0348a70c)
The model is the following:
```json
{
  "parent": "block/block",
  "textures": {
    "particle": "block/oak_leaves",
    "all": "block/oak_leaves",
    "1": "block/snow"
  },
  "elements": [
{
    "from": [0, 0, 0],
    "to": [16, 16, 16],
    "faces": {
      "north": {"uv": [0, 0, 16, 16], "texture": "#all", "cullface": "north", "tintindex": 0},
      "east": {"uv": [0, 0, 16, 16], "texture": "#all", "cullface": "east", "tintindex": 0},
      "south": {"uv": [0, 0, 16, 16], "texture": "#all", "cullface": "south", "tintindex": 0},
      "west": {"uv": [0, 0, 16, 16], "texture": "#all", "cullface": "west", "tintindex": 0},
      "up": {"uv": [0, 0, 16, 16], "texture": "#all", "cullface": "up", "tintindex": 0},
      "down": {"uv": [0, 0, 16, 16], "texture": "#all", "cullface": "down", "tintindex": 0}
    }
  },{
      "from": [0, 0, 0],
      "to": [16, 16, 16],
      "faces": {
        "north": {"uv": [0, 0, 16, 16], "texture": "#1", "cullface": "north"},
        "east": {"uv": [0, 0, 16, 16], "texture": "#1", "cullface": "east"},
        "south": {"uv": [0, 0, 16, 16], "texture": "#1", "cullface": "south"},
        "west": {"uv": [0, 0, 16, 16], "texture": "#1", "cullface": "west"},
        "up": {"uv": [0, 0, 16, 16], "texture": "#1", "cullface": "up"}
      }
    }]
}
```

**Comment by douira** [comment]
Possibly related to https://github.com/CaffeineMC/sodium/issues/2810?

**Comment by joe-vettek** [comment]
> Possibly related to #2810?

i think yes, but not need Continuity

**Comment by joe-vettek** [comment]
have you tried glass block?

**Comment by joe-vettek** [comment]
@jellysquid3  yes,  iris is not need in fact

**Comment by joe-vettek** [comment]
After testing version 0.6.9, I found that the problem seemed to be solved. However, I saw the comments in the code saying that this is a temporary solution.

**Comment by joe-vettek** [comment]
In the previous version, I later found a possible solution, which is to adjust the transparency of a pixel to semi-transparent, such as 80%, so that the entire texture can be covered smoothly. I don't know if it will help.

**Comment by jellysquid3** [comment]
> After testing version 0.6.9, I found that the problem seemed to be solved...

The patch was only to mitigate NeoForge block models that specified more than one render layer. Fabric mods and resource packs are not currently mitigated.

**Comment by joe-vettek** [comment]
https://github.com/CaffeineMC/sodium/commit/dc1d9138d753563efe8fb54aab63934b95754607

it's here

**Comment by jellysquid3** [comment]
Yes, I'm aware. I helped author that commit... But it wasn't _supposed_ to fix anything but NeoForge mods that have multiple render types specified to a block model, which resource packs don't have access to.

**Comment by joe-vettek** [comment]
oh well i got it. the main key is to disable allowDowngrade

**Comment by joe-vettek** [comment]
![Image](https://github.com/user-attachments/assets/0b3e0c6a-498a-4d87-9944-cb8a88a82902)
This is probably why older versions such as 0.6.5 can achieve correct coverage using textures with semi-transparent points, because they will not be downsampled.

**Comment by douira** [comment]
I think this and https://github.com/CaffeineMC/sodium/issues/2810 are essentially the same issue and we should close one of them. I'm for closing the other issue even though it is older because this issue demonstrates the ability to reproduce it even without Continuity, though it stems from the same effect.

Further, analysis of the code in this issue's comments is not required as we have already repeatedly discussed this issue on discord and are fairly confident about the mechanism.

**Comment by douira** [comment]
[Mojang has closed this as Won't Fix](https://mojira.dev/MC-306609), which means that resource pack authors will need to either use an alternative pack format provided by mods like Continuity or work around this change in some other way. We're not going to be implementing extensions to the resource pack format to fix this as that is out of scope.
